app/src/main/java/com/kylecorry/trail_sense/tools/metaldetector/ui/FragmentToolMetalDetector.kt

```diff
@@ -174,6 +195,19 @@ class FragmentToolMetalDetector : BoundFragment<FragmentToolMetalDetectorBinding
             haptics.off()
         }
     }
+    private fun updateMetalSoundIntensity(magneticField: Float) {
+        // manage
+        if (magneticField > lastMagneticField && volume < maxVolume) {
```

Comment:
Rather than increase the volume the longer it is exposed to the magnetic field, can you make it so it operates on the intensity above and below the threshold and use SolMath.map to get a constant volume for a given intensity.

Ex. If it is 5 above the threshold then it will be 5/x percent of the max volume, where x is the intensity of max volume (I'll say 30 for now, but will require experimentation)

Reply:
This might not be reliable approach because stable can be any number, depending on how close it is to the metal

the sound volume is from 0 min and 1 max 

this part is tricky, reason I stick with controlling the volume base on how the intensity increase / decrease

Reply:
I think that is acceptable for now - if you get closer to the metal and get a stronger reading it should be louder. To calculate the magnetic intensity, you should use the following algorithm:

If metal is not detected -> volume = 0
else -> volume = SolMath.map(delta - threshold, 0f, 30f, 0f, 1f, true), where delta uses the same math as this: https://github.com/kylecorry31/Trail-Sense/blob/d0575bf49ed06838b507eb0a937310881e513ae1/app/src/main/java/com/kylecorry/trail_sense/tools/metaldetector/ui/FragmentToolMetalDetector.kt#L188

This algorithm will map stronger magnetic interference to higher volumes. The part that can be adjusted with some experimentation is the 30f (the maximum difference between the magnetic field and the threshold boundary that equates to max volume).